wrapper/src/main/java/software/amazon/jdbc/hostavailability/ExponentialBackoffHostAvailabilityStrategy.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.hostavailability;
+
+import static software.amazon.jdbc.hostavailability.HostAvailabilityStrategyFactory.HOST_AVAILABILITY_STRATEGY_INITIAL_BACKOFF_TIME;
+import static software.amazon.jdbc.hostavailability.HostAvailabilityStrategyFactory.HOST_AVAILABILITY_STRATEGY_MAX_RETRIES;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Properties;
+import software.amazon.jdbc.util.Messages;
+
+public class ExponentialBackoffHostAvailabilityStrategy implements HostAvailabilityStrategy {
+
+  public static String NAME = "exponentialBackoff";
+  private int maxRetries = 5;
+  private int initialBackoffTimeSeconds = 30;
+  private int notAvailableCount = 0;
+  private Timestamp lastChanged;
+
+  public ExponentialBackoffHostAvailabilityStrategy(Properties props) {
```

Comment:
You need to register these configuration parameters (HOST_AVAILABILITY_STRATEGY_MAX_RETRIES and HOST_AVAILABILITY_STRATEGY_INITIAL_BACKOFF_TIME) in static constructor similar to HostMonitoringConnectionPlugin.

Reply:
Done!